File: core/umui_core/models/job.py
"""Domain model for UMUI jobs."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Job:
    """A job within an experiment in the UMUI database.

    Fields match the legacy .job file format (alternating field/value pairs).
    """

    job_id: str
    exp_id: str
    version: str
    description: str
    opened: str
    atmosphere: str
    ocean: str
    slab: str
    mesoscale: str

    def is_locked(self) -> bool:
        """Check if job is currently locked (opened) by someone."""
        return self.opened != "N" and self.opened != ""

    def locked_by(self) -> str | None:
        """Return the user who has this job locked, or None."""
        if self.is_locked():
            return self.opened
        return None

File: ui/src/types/job.ts
export interface JobResponse {
  readonly job_id: string;
  readonly exp_id: string;
  readonly version: string;
  readonly description: string;
  readonly opened: string;
  readonly atmosphere: string;
  readonly ocean: string;
  readonly slab: string;
  readonly mesoscale: string;
}

export interface JobListResponse {
  readonly jobs: readonly JobResponse[];
}

export interface CreateJobRequest {
  readonly job_id: string;
  readonly description?: string;
  readonly version?: string;
}

export interface UpdateJobRequest {
  readonly description?: string;
  readonly version?: string;
  readonly atmosphere?: string;
  readonly ocean?: string;
  readonly slab?: string;
  readonly mesoscale?: string;
}

export interface CopyJobRequest {
  readonly dest_exp_id: string;
  readonly dest_job_id: string;
  readonly description?: string;
}
